var/spack/repos/builtin/packages/ocamlbuild/package.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class Ocamlbuild(MakefilePackage):
+    """OCamlbuild is a generic build tool,
+       that has built-in rules for building OCaml library and programs."""
+
+    # Add a proper url for your package's homepage here.
+    homepage = "https://ocaml.org/learn/tutorials/ocamlbuild/"
+    url      = "https://github.com/ocaml/ocamlbuild/archive/0.14.0.tar.gz"
+    git      = "https://github.com/ocaml/ocamlbuild/ocamlbuild.git"
+
+    # Add a list of GitHub accounts to
+    # notify when the package is updated.
+    maintainers = ['scemama', 'cessenat']
+
+    # Add proper versions here.
+    # spack -d install --no-checksum ocamlbuild
+    version('master', sha256='bee0f96a265544818a4eb1a9cda0abd402ad160330f01a95229ba2d1011f4df3')
```

Comment:
I should note that it is rare -- as in only three built-in packages -- for a package to provide the `sha256` for `master`.  Typically you'll see

```
    version('master', branch='master')
```

or

```
    version('master')
```

though the latter *will* require use of `--no-checksum` to build at master since the `sha256` is not provided.  The reason is the hash for the `master` branch will likely change as a package evolves and adds dependencies, etc.

Reply:
@scemama What are your thoughts on providing a hash for this package?

Reply:
I agree with @tldahlgren . If you provide the hash, you say explicitly that you want a particular commit from the master branch. If you don't provide the hash, you will take the latest one.
So I would provide hashes for identified versions and no hash for master to avoid confusion and to be consistent with the other packages.

Reply:
Personally I would prefer a git checkout of the master branch instead of a checksum for some random commit. It will be a moving target, but we'll still have stable releases installed by default.

Reply:
Right, I remove the hash.